x-pack/legacy/plugins/reporting/server/browsers/network_policy.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as _ from 'lodash';
+import { parse } from 'url';
+
+interface FirewallRule {
+  allow: boolean;
+  host?: string;
+  protocol?: string;
+}
+
+const isHostMatch = (actualHost: string, ruleHost: string) => {
+  const hostParts = actualHost.split('.').reverse();
+  const ruleParts = ruleHost.split('.').reverse();
+
+  return _.every(ruleParts, (part, idx) => part === hostParts[idx]);
+};
+
+export const allowRequest = (url: string, rules: FirewallRule[]) => {
+  const parsed = parse(url);
+
+  if (!rules.length) {
+    return true;
+  }
+
+  // Accumulator has three potential values here:
+  // True => allow request, don't check other rules
+  // False => reject request, don't check other rules
+  // Undefined => Not yet known, proceed to next rule
+  const allowed = rules.reduce((result: boolean | undefined, rule) => {
+    if (typeof result === 'boolean') {
+      return result;
+    }
+
+    const hostMatch = rule.host ? isHostMatch(parsed.host || '', rule.host) : true;
+
+    const protocolMatch = rule.protocol ? parsed.protocol === rule.protocol : true;
+
+    const isRuleMatch = hostMatch && protocolMatch;
+
+    return isRuleMatch ? rule.allow : undefined;
+  }, undefined);
+
+  return typeof allowed === 'undefined' || allowed;
```

Comment:
```suggestion
  return typeof allowed === 'undefined' ? false : allowed;
```

When the user has defined rules, but none of them match, it'd be safest for us to deny the request.

I'd recommend adding this test as well:
```
  it('denies requests when no rules match', () => {
    const url = 'https://not-kibana.com/cool/route/bro';
    const rules = [{ allow: true, host: 'kibana.com' }];

    expect(allowRequest(url, rules)).toEqual(false);
  });
```